aptihramy-backend/api.py

```diff
@@ -100,21 +100,33 @@ def get_tracking_chain(tracker_id: int, db: Database = Depends(get_database)):
 
 @app.get("/materialized_frames")
 def get_materialized_frames(tracker_id: int, db: Database = Depends(get_database)):
-    try:
-        materialized_chain = db.get_materialized_tracking_chain(tracker_id)
-    except:
-        raise HTTPException(
-            status_code=400,
-            detail=f"Unkwown tracking chain ${tracker_id}",
-        )
-    return MaterializedTrackingChainModel.from_materialized_tracking_chain(
+    materialized_chain = db.get_materialized_tracking_chain(tracker_id)
+    frame_idx_rec_idxs = db.get_frame_idx_record_idxs_from_materialized_chain(
         materialized_chain
     )
 
+    # frame idx -> record idx -> raw values [value_of_feature1, value_of_feature2)
+    raw_values: dict[tuple[int, int], list[Element]] = {}
+    # frame idx -> record idx -> raw values [value_of_feature1, value_of_feature2)
+
+    normalized_values: dict[tuple[int, int], list[Element]] = {}
+    for frame_idx, record_idxs in frame_idx_rec_idxs.items():
+        for record_idx in record_idxs:
+            raw_values[(frame_idx, record_idx)] = (
+                db.get_raw_values_for_frame_idx_record_idx(frame_idx, record_idx)
+            )
+            normalized_values[(frame_idx, record_idx)] = (
+                db.get_normalized_values_for_frame_idx_record_idx(frame_idx, record_idx)
+            )
```

Comment:
I'm not sure I understand this part, isn't the record/normalized record already stored in `MaterializedTrackerFrame.record` and `MaterializedTrackerFrame.normalized_record` ?

Reply:
Make sense